### PR TITLE
Add a new workflow to create and publish the SBOM.

### PR DESCRIPTION
**Pull request by jcouture**
## 📖 Description and motivation

Creating a new workflow, triggering upon a push on the main branch, which will publish the [SBOM](https://www.aquasec.com/cloud-native-academy/supply-chain-security/sbom/) to the GitHub Dependency submission API. The scan lists relevant information from the Docker Image system and libraries as well as the Ruby Gems and NPM Packages.

## 🦀 Dispatch

`#dispatch/rails`

### CODE REVIEW

File: .github/workflows/upload-sbom.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build Docker image
+        run: make build
+
+      - uses: anchore/sbom-action@latest
+        with:
+          image: 'mirego/killswitch:${{ github.sha }}'
+          dependency-snapshot: true
```
Comment (remi) [inline .github/workflows/upload-sbom.yaml:24]: We should use spaces, not tabs 🙂

File: .github/workflows/upload-sbom.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    runs-on: ubuntu-20.04
+    permissions:
+      contents: read
+      security-events: write
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build Docker image
+        run: make build
```
Comment (remi) [inline .github/workflows/upload-sbom.yaml:19]: We should provide a custom `DOCKER_TAG` and then use in the image name below.

https://github.com/mirego/killswitch/blob/main/Makefile#LL4C5-L4C5

File: .github/workflows/upload-sbom.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+  upload-sbom:
+    runs-on: ubuntu-20.04
+    permissions:
+      contents: read
+      security-events: write
+
+    steps:
+      - uses: actions/checkout@v2
```
Comment (remi) [inline .github/workflows/upload-sbom.yaml:16]: We should use `v3`

```suggestion
      - uses: actions/checkout@v3
```

I’ll fix the other workflows!

File: .github/workflows/upload-sbom.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+name: Syft SBOM Creation
```
Comment (remi) [inline .github/workflows/upload-sbom.yaml:1]: The name should match the workflow filename.

I suggest:

```suggestion
# .github/workflows/sbom.yaml
name: SBOM
```

What do you think?

File: .github/workflows/upload-sbom.yaml (hunk 1 above)
Comment (remi) [reply .github/workflows/upload-sbom.yaml:24]: Nevermind, github.com’s UI was broken 😬

File: .github/workflows/upload-sbom.yaml (hunk 4 above)
Comment (jcouture) [reply .github/workflows/upload-sbom.yaml:1]: Sure, let’s do it!

File: .github/workflows/upload-sbom.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Build Docker image
+        run: make build
+
+      - uses: anchore/sbom-action@latest
+        with:
+          image: 'mirego/killswitch:${{ github.sha }}'
+          dependency-snapshot: true
```
Comment (remi) [inline .github/workflows/upload-sbom.yaml:25]:
```suggestion
      - run: make build
        env:
          DOCKER_TAG: sbom-snapshot

      - uses: anchore/sbom-action@latest
        with:
          image: mirego/killswitch:sbom-snapshot
          dependency-snapshot: true
```

File: .github/workflows/upload-sbom.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# .github/workflows/sbom.yaml
```
Comment (remi) [inline .github/workflows/upload-sbom.yaml:1]: Well that was supposed to be an indication to rename the file, not add this line. Sorry 🙂

Comment (jcouture) [reply .github/workflows/upload-sbom.yaml]: 👍

**Review by remi — APPROVED** [review]
🙌

**Pull request merged**